fix(lodash): lodash methods added in plain js/ts

PunitSoniME opened the pull request:
<!--
Try to use a template, found at .github/PULL_REQUEST_TEMPLATE/
[here](https://github.com/typegoose/typegoose/tree/master/.github/PULL_REQUEST_TEMPLATE)
please don't forget to click on raw, and copy that, not the already "compiled" one
-->

<!--
## Make sure you have done these steps

- Make sure you have Read & followed these steps in [CONTRIBUTING](https://github.com/typegoose/typegoose/tree/master/.github/CONTRIBUTING.md)
- remove the parts that are not applicable
- Please have "Allow edits from maintainers" activated
-->

<!--Write your PR description here (above "Related Issues")-->

## Related Issues

<!--add "fixes" / "closes" before an number to indicate that these will be fixed by this pr-->

- <!--Add the issue this pr fixes here, in the style of "#1"-->

hasezoey commented:
could you explain why this change would be necessary, also is there a performance difference between this implementation and lodash's?

originally i was going either "no lodash" or "all lodash" (no small packages), because from what i could tell, the "lodash" package was used more (and so is installed more) than the smaller packages and so less dependencies

PunitSoniME commented:
Time difference of lodash's omit and the code which I added for omit ( example is taken from lodash's documentation )

![image](https://user-images.githubusercontent.com/11643426/233074044-04f9c064-f8c1-4d0d-99f5-da0018a82c78.png)


Time difference of lodash's intersection and the code which I added for intersection ( example is taken from lodash's documentation )

![image](https://user-images.githubusercontent.com/11643426/233074689-84b4bfbf-117a-40fb-a7b9-32e8ae3e596d.png)


If you think this changes are not necessary, we can ignore it and close this PR 👍

hasezoey reviewed (commented):
i will keep the current way for now, but may consider merging this in the future for a major version or something

PunitSoniME commented on src/internal/utils.ts:
```diff
@@ -1,4 +1,4 @@
-import { intersection, mergeWith, omit } from 'lodash';
+import mergeWith = require('lodash/mergeWith');
```
Yes, if we make it like `import mergeWith from 'lodash.mergeWith'` it gives an error in build

hasezoey replied:
shouldnt `lodash/mergeWith` just work, or is this just valid if it a path *inside* a package?

PunitSoniME replied:
`lodash/mergeWith` works when `lodash` is installed

I uninstalled whole lodash package and installed only mergeWith library `lodash.mergeWith` to reduce the size

hasezoey replied:
then unless this function can also be de-lodash'ed, i dont think i will go to smaller packages, [because of this quote](https://lodash.com/per-method-packages):

> However, use of these packages is discouraged and they [will be removed in v5](https://github.com/lodash/lodash/issues/3838#issuecomment-398592530).
